packages/docusaurus/src/webpack/utils.ts

```diff
@@ -139,14 +140,19 @@ export function getBabelOptions({
 // we want to support multiple js loader implementations (babel + esbuild)
 export function getJSLoader({
   isServer,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   babelOptions,
 }: {
   isServer: boolean;
   babelOptions?: TransformOptions | string;
 }): RuleSetRule {
   return {
-    loader: require.resolve('babel-loader'),
-    options: getBabelOptions({isServer, babelOptions}),
+    loader: require.resolve('esbuild-loader'),
+    options: {
+      loader: 'tsx',
+      format: isServer ? 'cjs' : undefined,
+      target: 'es2017',
```

Comment:
maybe target node12 for isServer would more appropriate?

For browsers, it seems we can't use esbuild directly but it seems there is a way to convert browserlists to esbuild target: https://github.com/evanw/esbuild/issues/121